server/src/main/java/com/flipkart/varadhi/utils/SubscriptionHelper.java

```diff
@@ -0,0 +1,62 @@
+package com.flipkart.varadhi.utils;
+
+import com.flipkart.varadhi.entities.SubscriptionResource;
+import com.flipkart.varadhi.entities.SubscriptionShard;
+import com.flipkart.varadhi.entities.VaradhiSubscription;
+
+import static com.flipkart.varadhi.entities.VersionedEntity.NAME_SEPARATOR;
+import static com.flipkart.varadhi.entities.VersionedEntity.NAME_SEPARATOR_REGEX;
+
+public final class SubscriptionHelper {
+    private SubscriptionHelper() {
+    }
+
+    private static final SubscriptionShard[] dummyShards =
+            new SubscriptionShard[]{new SubscriptionShard(0, null, null, null)};
+
+    public static VaradhiSubscription fromResource(SubscriptionResource subscriptionResource, int version) {
+        return new VaradhiSubscription(
+                buildSubscriptionName(subscriptionResource.getProject(), subscriptionResource.getName()),
+                version,
+                subscriptionResource.getProject(),
+                buildTopicName(subscriptionResource.getTopicProject(), subscriptionResource.getTopic()),
+                subscriptionResource.getDescription(),
+                subscriptionResource.isGrouped(),
+                subscriptionResource.getEndpoint(),
+                subscriptionResource.getRetryPolicy(),
+                subscriptionResource.getConsumptionPolicy(),
+                dummyShards // fixme: this is a hack to make the flow work
```

Comment:
due to checks on shards, I need to provide some dummy values here
in future when shards flow is working then we can fix this code